File: pinyougou-shop-web/src/main/resources/static/js/service/goodsService.js
//服务层
app.service('goodsService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../shopGoods/findAll');
    }
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../shopGoods/findPage?page=' + page + '&rows=' + rows);
    }
    //分页
    this.findPage2 = function (page, rows) {
        return $http.get('../shopGoods/findPage2?page=' + page + '&rows=' + rows);
    }
    //查询实体
    this.findOne = function (id) {
        return $http.get('../shopGoods/findOne?id=' + id);
    }
    //增加
    this.add = function (entity) {
        return $http.post('../shopGoods/add', entity);
    }
    //增加
    this.saveGoods = function (entity) {
        return $http.post('../shopGoods/saveGoods', entity);
    }
    //修改
    this.update = function (entity) {
        return $http.post('../shopGoods/update', entity);
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../shopGoods/delete?ids=' + ids);
    }
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../shopGoods/search?page=' + page + "&rows=" + rows, searchEntity);
    }
    //状态更新
    this.updateStatus = function (sign, status, ids) {
        return $http.get('../shopGoods/UpDownGoods?ids=' + ids + "&status=" + status + "&sign=" + sign);
    }
//删除
    this.findGoods = function (id) {
        return $http.get('../shopGoods/findGoods?id=' + id);
    }

});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/service/GoodsService.java
package com.pinyougou.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.Goods;
import com.pinyougou.pojo.TbGoods;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface GoodsService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbGoods> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage2(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(TbGoods goods);

    /**
     * 增加
     */
    public void addGoods(Goods goods);

    //更新
    public void updateGoods(Goods goods);

    /**
     * 修改
     */
    public void update(TbGoods goods);

    public void updateByPrimaryKeySelective(TbGoods goods);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbGoods findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbGoods goods, int pageNum, int pageSize);

    /**
     * 查询good集合
     *
     * @param id
     * @return
     */
    public Goods findGoods(Long id);
}
